yt_dlp/extractor/yappy.py

```diff
@@ -97,3 +98,39 @@ def _real_extract(self, url):
             'categories': traverse_obj(media_data, ('categories', ..., 'name')) or None,
             'repost_count': int_or_none(media_data.get('sharingCount'))
         }
+
+
+class YappyProfileIE(InfoExtractor):
+    _VALID_URL = r'https?://yappy\.media/profile/(?P<id>\w+)'
+    _TESTS = [{
+        'url': 'https://yappy.media/profile/59a0c8c485e5410b9c43474bf4c6a373',
+        'info_dict': {
+            'id': '59a0c8c485e5410b9c43474bf4c6a373',
+            'title': 'profile',
+        },
+        'playlist_count': 16,
+    }]
+
+    def _real_extract(self, url):
+        profile_id = self._match_id(url)
+
+        entries = []
+        for page_num in itertools.count(1):
+            profile_data = self._download_json(f'https://yappy.media/api/video/list/{profile_id}?page={page_num}',
+                                               note='Downloading API JSON',
+                                               errnote='Unable to download API page',
+                                               video_id=f'{profile_id} page {page_num}')
+
+            if not profile_data['results']:
+                break
+
+            entries.extend([
+                self.url_result(
+                    f'https://yappy.media/video/{video["uuid"]}',
+                    ie=YappyIE.ie_key(),
+                    video_id=video['uuid'],
+                    video_title=video.get('description') or video['uuid'])
+                for video in profile_data['results']
+            ])
+
+        return self.playlist_result(entries, profile_id)
```

Comment:
It's better to use a generator for this.
You can use `traverse_obj` to safely traverse the JSON object and filter for dicts with a `uuid` key
```suggestion
        def entries(profile_id):
            for page_num in itertools.count(1):
                videos = traverse_obj(self._download_json(
                    f'https://yappy.media/api/video/list/{profile_id}?page={page_num}',
                    profile_id, f'Downloading profile page {page_num} JSON'),
                    ('results', lambda _, v: v['uuid']))
                if not videos:
                    break
                for video in videos:
                    yield self.url_result(
                        f'https://yappy.media/video/{video["uuid"]}', YappyIE,
                        video['uuid'], video.get('description'))

        return self.playlist_result(entries(profile_id), profile_id)
```